File: kotlin/sprint5/E/Solution.kt
/**
 * Comment it before submitting
 *
 * class Node {
 *     var value: Int
 *     var left: Node?
 *     var right: Node?
 *
 *     constructor(value: Int) {
 *         this.value = value
 *         right = null
 *         left = null
 *     }
 *
 *     constructor(value: Int, left: Node?, right: Node?) {
 *         this.value = value
 *         this.left = left
 *         this.right = right
 *     }
 * }
 */

fun treeSolution(head: Node?): Boolean {
    // Your code
    // “ヽ(´▽｀)ノ”
}

fun test() {
    val node1 = Node(1, null, null)
    val node2 = Node(4, null, null)
    val node3 = Node(3, node1, node2)
    val node4 = Node(8, null, null)
    val node5 = Node(5, node3, node4)
    assert(treeSolution(node5))
    node2.value = 5
    assert(!treeSolution(node5))
}

File: swift/sprint5/E/solution.swift
/*
 Comment it before submitting

 open class Node {
     
     public var value: Int
     public var left: Node?
     public var right: Node?
     
     public init(_ value: Int) {
         self.value = value
         self.right = nil
         self.left = nil
     }
 }
 */

func solution(node: Node?) -> Bool {
    // Your code
    // ヽ(´▽`)/
    return false
}


func test() {
    var node1 = Node(1)
    var node2 = Node(4)
    var node3 = Node(3)
    node3.left = node1
    node3.right = node2
    var node4 = Node(8)
    var node5 = Node(5)
    node5.left = node3
    node5.right = node4
    assert(solution(node: node5))
    node2.value = 5
    assert(!solution(node: node5))
}
